allo/ir/builder.py

```diff
@@ -201,8 +201,74 @@ def build_all_for(ctx, node, attr):
         is_affine = True
         iter_args = node.iter.args
         if attr in {"grid", "reduction"}:
-            grid = [ASTResolver.resolve_constant(x, ctx) for x in iter_args]
-            for_loops = build_for_loops(grid, ctx.get_ip(), names, stage_name)
+            for_loops = []
+            if stage_name is None:
+                stage_name = "S_" + "_".join(names)
+
+            ip_handle = ctx.get_ip()
+            for ind, arg in enumerate(iter_args):  # Traversal
```

Comment:
Did you copy the following code from the other branch? Maybe create a common function to reduce duplicated code